### DISCUSSION TITLE
Max speed extension in exported gpx file

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#549] [Q&A] [answered]
Just being curious - I saw a few new extention definition in Cruiser gpx output:

![2024-05-25 11_06_38-D__Dropbox_Test Semptquelle_cn gpx - Notepad++](https://github.com/devemux86/cruiser/assets/121806238/1f2f3666-69c0-481d-842b-e0799b8e8f6a)

This is the whole file: [Test Semptquelle_cn.gpx.txt](https://github.com/devemux86/cruiser/files/15442442/Test.Semptquelle_cn.gpx.txt)

Could you reveal the secret?

**Comment by devemux86** [comment, marked as answer]
@WolfgangZ42 

It is part of the GPX navigation track (offline) format for saving complete routes with their max speeds.
(so we can load them again offline without route calculation)

- https://github.com/devemux86/cruiser/discussions/250

The GPX 1.1 specification no longer defines a speed tag, so we need to use an extension.

**Reply by devemux86** [reply]
In this route example, BRouter probably does not return max speed for the most road sections
and it returns empty max speed for some sections at the end.
